src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -88,6 +89,13 @@ export class NotebookLinkHandler {
 			 * We return the absolute path for the link so that it will get used in the as the href for the anchor HTML element
 			 * (in linkCalloutDialog document.execCommand('insertHTML') and therefore will call getLinkURL() with HTMLAnchorElement to then get the relative path
 			*/
+			// Need to encode URI here in order for user to click the proper encoded link in WYSIWYG
+			// skip encoding it if it's already encoded
+			let encodedLinkURL = this._isFile ? escapeUrl(this._link) : this._link;
+			if (encodedLinkURL !== decodeURI(encodedLinkURL)) {
```

Comment:
This check still confuses me. A couple questions : 

1. When would the link already be encoded? I tried various things and no matter what I did it always seemed to be the unencoded version
2. Why are you calling escapeUrl just for files? What's special about them that requires that extra decoding?
3. On line 86 we return early if it's a relative path - shouldn't we also be doing this same thing regardless of whether it's absolute or relative?

@VasuBhog You did the original change - do you see any potential issues with this? I tested it out and it does seem to work correctly but there's a lot of things to test here...

Reply:
I believe that this change would work however it can be consolidated since we should be using encodeURI instead of escapeURL from my understanding as that would work for this condition as shown below for file paths and other links: 
`encodedLinkURL` - escapeURL(this._link)
<img width="680" alt="Screen Shot 2021-12-02 at 2 02 07 PM" src="https://user-images.githubusercontent.com/23587151/144512071-abc40a77-7b73-4190-a4ec-b3511f5c3fa7.png">